### ISSUE TITLE
Final Project Peer Review

### ISSUE THREAD

**Issue by kingofguan** [ZRich098/ORIE-4741#13]
This project predicts the significance of emergency calls based on a number of call features. The dataset includes a wide range of descriptive features such as timestamp, location, severity assessment, etc. The team uses the data to classify and predict the severity finally assigned to the situation the call referred to.

**Areas of Strengths:**

- The team did great data preprocessing, identified problems, and acquired relevant additional data that improved the outcome. Specifically, the team applied their knowledge about the real-world situation. The team imported more demographic data that improved the results by capturing communities' size and the relevant emergency resources they require.

- The team used Random Forest on top of the linear approximation methods. In this problem, I believe the random forest model could work better as it assimilates the process of handing the call. As a result, the outcome could be more actionable as the operator would receive a call by walking through a tree of questions and take further action according to the response. Also, it is good to use a grid search to fine-tune hyperparameter as the team did.

- The team explored more models in the AutoML part. The PCA application is very impressive. It can help identify what features are more critical and requires the authority’s attention. Furthermore, I imagine the emergency call data would have many missing values in the real-world, and methods such as PCA could potentially resolve those issues.

**Areas for improvements/Advice:**

- I think setting a clear and more detailed performance benchmark could further improve the project. For instance, the best model has a 0.31 error rate and 178s in calculating response time. For minor situations at a far-away location, these outcomes could be excellent. However, if it is a fire hazard or serious offense, 0.31 and 178s might not be optimal. Therefore, I think the project could be improved if the team could tune the model and prioritize incidents with certain characteristics, such as high severity.

- Although not being the primary focus of the project, I would like to learn why the team chose Ridge Regression, Gradient Boosting, K-Neighbors over others, and what fundamental facts about the data result in the difference in performance.

- If I were part of the team, one further analysis is error analysis between the “Initial severity level code” and the “Final severity level code”. This analysis could help study the mistake a well-trained operator would make. The operator’s decision-making knowledge and experience are great models that could be used to calibrate the team’s model.

I learned a lot from your work. Have a great holiday!
